### PR TITLE
Lessen breakup kicks and more

### LINKED ISSUE #178: Block Spam Message
**Is your feature request related to a problem? Please describe.**
My phone keeps getting spam messages and the notification sound of spam messages is very annoying. And constant spam messages in chat.

**Describe the solution you'd like**
Blocking spam messages.

**Provide Reasoning**
Spam messages are annoying and notifications from spam messages are a headache. And seeing spam messages constantly in chat is eye catching, so I want to block spam messages.

**Additional context**
Someone from the forum sent me these source code. Can review this? I don't know coding. Thanks.
Source Code: https://github.com/HolyWurl/Block-GTASpamMessage

### LINKED ISSUE #234: Expand Remote features to allow multiple players
**Is your feature request related to a problem? Please describe.**
Currently remote features can only be enabled on the selected player. 

**Describe the solution you'd like**
A system to allow remote features for more than one player at once.

**Provide Reasoning**
You might want to grant "Never wanted" for 2 players at once (as an example).

**Additional context**
A struct could be created containing bools that specify whether a remote function should be executed on a certain player. This struct could be a member of each players own struct.

### LINKED ISSUE #620: [Bug]: Crash of injection
### Describe the bug

Crash of injection

### Steps To Reproduce

[22:32:48][INFO/main.cpp:58]: Yim's Menu Initializing
Found 'SCREEN_RESOLUTION' GTA5.exe+0x16C157
[22:32:48][INFO/main.cpp:61]: Settings Loaded.
Found 'RC' GTA5.exe+0x13C825C
Found 'MWL' GTA5.exe+0x5D45BB
Found 'GS' GTA5.exe+0x2BF068
Found 'ISA' GTA5.exe+0xEC43
Found 'PF' GTA5.exe+0x728E48
Found 'NPM' GTA5.exe+0x1076C6C
Found 'NH' GTA5.exe+0x15C2131
Found 'FV' GTA5.exe+0x15BF0E0
Found 'ST' GTA5.exe+0x15C2087
Found 'SP' GTA5.exe+0xA3B7CD
Found 'SG' GTA5.exe+0x15C19D8
Found 'CGSHM' GTA5.exe+0x15C6B69
Found 'S' GTA5.exe+0x1345189
Found 'MSB' GTA5.exe+0xA395B2
Found 'WMSB' GTA5.exe+0xCDCE99
Found 'NRF' GTA5.exe+0x4A0C32
Found 'GTS' GTA5.exe+0x15C1BA0
Found 'TK' GTA5.exe+0xA3F898
Found 'TSE' GTA5.exe+0xD6A848
Found 'REH' GTA5.exe+0x167940C
Found 'SEA' GTA5.exe+0x1664215
Found 'RBWD' GTA5.exe+0x109D111
Found 'RBA' GTA5.exe+0x12E353C
Found 'RBS' GTA5.exe+0x1170CFC
Found 'RBB' GTA5.exe+0x11278C7
Found 'WBD' GTA5.exe+0x1127E10
Found 'WBQ' GTA5.exe+0x12EB2EC
Found 'WBI64' GTA5.exe+0x1174381
Found 'WBI32' GTA5.exe+0x1173253
Found 'WBB' GTA5.exe+0x1172DD8
Found 'WBA' GTA5.exe+0x1663FD6
Found 'WPGSDN' GTA5.exe+0x10CCE94
Found 'RCOE-Patch' GTA5.exe+0xCCC264
Found 'RI' GTA5.exe+0x1E9C
Found 'PTH' GTA5.exe+0x15C6625
Found 'GSH' GTA5.exe+0x15C7AF4
Found 'BE' GTA5.exe+0xA64FBB
Found 'SNITL' GTA5.exe+0x13BC4BA
Found 'CNOM' GTA5.exe+0xB108C4
Found 'NPMI' GTA5.exe+0x10D78E7
Found 'NPMS' GTA5.exe+0x110BB3A
Found 'FR' GTA5.exe+0xCC8F38
Found 'GSCFWC' GTA5.exe+0xA3A934
Found 'GGCC' GTA5.exe+0xA6F1F2
Found 'GPR' GTA5.exe+0x118F304
Found 'WPGDN' GTA5.exe+0x10CD9B0
Found 'RNM' GTA5.exe+0x14035FD
Found 'GNED' GTA5.exe+0x19D32F8
Found 'API' GTA5.exe+0x167F4D8
Found 'RCC' GTA5.exe+0x10E1EAC
Found 'CAD' GTA5.exe+0x16675F4
Found 'RCS/GSTFT/GNOFP/GNO/GSTI' GTA5.exe+0x10E2E61
Found 'RBIST' GTA5.exe+0x10E202C
Found 'MHT' GTA5.exe+0xAF80
Found 'GLT' GTA5.exe+0xD86917
Found 'MCF' GTA5.exe+0x163D5E2
Found 'N' GTA5.exe+0x10448B1
Found 'RENC' GTA5.exe+0x142900A
Found 'FDGD' GTA5.exe+0x12CF527
Found 'FDS' GTA5.exe+0x12D2084
Found 'FPFC' GTA5.exe+0x12CBEB2
Found 'FPFOA' GTA5.exe+0x12D0DA0
Found 'FPFM' GTA5.exe+0x12D22FA
Found 'FPFUM' GTA5.exe+0x69E6C7
Found 'GVOV' GTA5.exe+0x2E0438
Found 'IMCD' GTA5.exe+0x53443D
Found 'SCP' GTA5.exe+0x2ED858
Found 'SCM' GTA5.exe+0x1088BD9
Found 'FMFS' GTA5.exe+0x13BB9D8
Found 'SGSBGH' GTA5.exe+0x439B89
Found 'JSBI' GTA5.exe+0x114EA64
Found 'INT' GTA5.exe+0x15BE806
Found 'VM' GTA5.exe+0x15C19E2
Found 'GU' GTA5.exe+0x13EC4BE
Found 'HT' GTA5.exe+0x13C1BF5
Found 'PGI' GTA5.exe+0x13DCF98
Found 'PIGI' GTA5.exe+0x1082ACB
Found 'C' GTA5.exe+0xCCAF27
Found 'SDEGSDN' GTA5.exe+0x10EC0C4
Found 'SPIDN' GTA5.exe+0x10EE41C
Found 'SVGDN' GTA5.exe+0x10F5B64
Found 'GVGAS' GTA5.exe+0x10CFE54
Found 'HJR' GTA5.exe+0x115341C
Found 'WJRD' GTA5.exe+0x1154088
Found 'QP' GTA5.exe+0x1421C2E
Found 'SGS' GTA5.exe+0x1166CBC
Found 'PD' GTA5.exe+0x70558
[22:32:50][DEBUG/module.cpp:49]: Waiting for socialclub.dll...
[22:32:50][INFO/main.cpp:64]: Pointers initialized.
[22:32:51][INFO/main.cpp:67]: Renderer initialized.
[22:32:51][INFO/main.cpp:70]: Fiber pool initialized.
[22:32:52][INFO/main.cpp:73]: Hooking initialized.
[22:32:52][INFO/main.cpp:76]: Thread pool initialized.
[22:32:52][DEBUG/thread_pool.cpp:21]: Allocating 3 threads in thread pool.
2022/11/19 22:32:52 848043

***** FATAL SIGNAL RECEIVED ******* 

***** Received fatal signal SIGABRT(22)	PID: 5420

*******	STACKDUMP *******

Loaded Modules:

C:\Program Files (x86)\Steam\gameoverlayrenderer64.dll Base Address: 0x7FFBB86B0000 Size: 0x188000
C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\GFSDK_ShadowLib.win64.dll Base Address: 0x7FFBB27D0000 Size: 0x3CC000
C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\GFSDK_TXAA_AlphaResolve.win64.dll Base Address: 0x7FFBC7200000 Size: 0x1C000
C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\GTA5.exe Base Address: 0x7FF64E1D0000 Size: 0x3D20600
C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\bink2w64.dll Base Address: 0x7FFBB7600000 Size: 0x8D000
C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\steam_api64.dll Base Address: 0x65930000 Size: 0x3F000
C:\Program Files (x86)\Steam\steamclient64.dll Base Address: 0x7FFBB1070000 Size: 0x157D000
C:\Program Files (x86)\Steam\tier0_s64.dll Base Address: 0x7FFBB83E0000 Size: 0x195000
C:\Program Files (x86)\Steam\vstdlib_s64.dll Base Address: 0x7FFBB8150000 Size: 0xB2000
C:\Program Files\Rockstar Games\Social Club\chrome_elf.dll Base Address: 0x7FFBB7870000 Size: 0x101000
C:\Program Files\Rockstar Games\Social Club\libcef.dll Base Address: 0x7FFB959E0000 Size: 0x809F000
C:\Program Files\Rockstar Games\Social Club\socialclub.dll Base Address: 0x7FFBB0650000 Size: 0xA14000
C:\Users\Administrator\Desktop\YimMenu (8).dll Base Address: 0x7FFBAE680000 Size: 0x1FA000
C:\Windows\SYSTEM32\AUDIOSES.DLL Base Address: 0x7FFBC2340000 Size: 0x181000
C:\Windows\SYSTEM32\CRYPTBASE.DLL Base Address: 0x7FFBD5180000 Size: 0xC000
C:\Windows\SYSTEM32\CRYPTSP.dll Base Address: 0x7FFBD5160000 Size: 0x18000
C:\Windows\SYSTEM32\CRYPTUI.dll Base Address: 0x7FFBC7480000 Size: 0x51000
C:\Windows\SYSTEM32\CoreMessaging.dll Base Address: 0x7FFBD2E50000 Size: 0xF2000
C:\Windows\SYSTEM32\CoreUIComponents.dll Base Address: 0x7FFBD2770000 Size: 0x35E000
C:\Windows\SYSTEM32\D3D10_1.DLL Base Address: 0x7FFBB9230000 Size: 0x31000
C:\Windows\SYSTEM32\D3D11.DLL Base Address: 0x7FFBD1D50000 Size: 0x264000
C:\Windows\SYSTEM32\D3DCOMPILER_43.dll Base Address: 0x7FFBB50A0000 Size: 0x26F000
C:\Windows\SYSTEM32\D3DCOMPILER_47.dll Base Address: 0x7FFBD17A0000 Size: 0x45D000
C:\Windows\SYSTEM32\DEVOBJ.dll Base Address: 0x7FFBD5560000 Size: 0x34000
C:\Windows\SYSTEM32\DINPUT8.dll Base Address: 0x7FFBC6AD0000 Size: 0x45000
C:\Windows\SYSTEM32\DNSAPI.dll Base Address: 0x7FFBD4CA0000 Size: 0xCB000
C:\Windows\SYSTEM32\DSOUND.dll Base Address: 0x7FFBB7AC0000 Size: 0x9C000
C:\Windows\SYSTEM32\DWrite.dll Base Address: 0x7FFBC8B20000 Size: 0x283000
C:\Windows\SYSTEM32\ESENT.dll Base Address: 0x7FFBC3C40000 Size: 0x32B000
C:\Windows\SYSTEM32\GLU32.dll Base Address: 0x7FFBC6B30000 Size: 0x2C000
C:\Windows\SYSTEM32\HID.DLL Base Address: 0x7FFBD40E0000 Size: 0xD000
C:\Windows\SYSTEM32\IPHLPAPI.DLL Base Address: 0x7FFBD4C60000 Size: 0x3B000
C:\Windows\SYSTEM32\MF.dll Base Address: 0x7FFBC5B40000 Size: 0x84000
C:\Windows\SYSTEM32\MFCORE.DLL Base Address: 0x7FFBB4330000 Size: 0x491000
C:\Windows\SYSTEM32\MFPlat.DLL Base Address: 0x7FFBBBA90000 Size: 0x1BC000
C:\Windows\SYSTEM32\MFReadWrite.dll Base Address: 0x7FFBB7C10000 Size: 0x10A000
C:\Windows\SYSTEM32\MSACM32.dll Base Address: 0x7FFBC5A80000 Size: 0x1E000
C:\Windows\SYSTEM32\MSASN1.dll Base Address: 0x7FFBD53A0000 Size: 0x12000
C:\Windows\SYSTEM32\MSVCP140.dll Base Address: 0x7FFBC7A10000 Size: 0x91000
C:\Windows\SYSTEM32\MSWSOCK.dll Base Address: 0x7FFBD4F70000 Size: 0x6A000
C:\Windows\SYSTEM32\NETAPI32.dll Base Address: 0x7FFBC97D0000 Size: 0x18000
C:\Windows\SYSTEM32\NTASN1.dll Base Address: 0x7FFBD5240000 Size: 0x3B000
C:\Windows\SYSTEM32\OLEACC.dll Base Address: 0x7FFBC03A0000 Size: 0x66000
C:\Windows\SYSTEM32\OPENGL32.dll Base Address: 0x7FFBB5740000 Size: 0x125000
C:\Windows\SYSTEM32\PROPSYS.dll Base Address: 0x7FFBD0C90000 Size: 0xF6000
C:\Windows\SYSTEM32\RTWorkQ.DLL Base Address: 0x7FFBBBA60000 Size: 0x30000
C:\Windows\SYSTEM32\SAMCLI.DLL Base Address: 0x7FFBCAC30000 Size: 0x19000
C:\Windows\SYSTEM32\SSPICLI.DLL Base Address: 0x7FFBD56F0000 Size: 0x31000
C:\Windows\SYSTEM32\Secur32.dll Base Address: 0x7FFBC9260000 Size: 0xC000
C:\Windows\SYSTEM32\UIAutomationCore.DLL Base Address: 0x7FFBBD3E0000 Size: 0x2F5000
C:\Windows\SYSTEM32\UMPDC.dll Base Address: 0x7FFBD5680000 Size: 0x12000
C:\Windows\SYSTEM32\USERENV.dll Base Address: 0x7FFBD5730000 Size: 0x2E000
C:\Windows\SYSTEM32\USP10.dll Base Address: 0x7FFBD0940000 Size: 0x19000
C:\Windows\SYSTEM32\VCRUNTIME140.dll Base Address: 0x7FFBC99A0000 Size: 0x19000
C:\Windows\SYSTEM32\VCRUNTIME140_1.dll Base Address: 0x7FFBC7EC0000 Size: 0xC000
C:\Windows\SYSTEM32\VERSION.dll Base Address: 0x7FFBCD620000 Size: 0xA000
C:\Windows\SYSTEM32\WINHTTP.dll Base Address: 0x7FFBD0750000 Size: 0x10C000
C:\Windows\SYSTEM32\WININET.dll Base Address: 0x7FFBC9DE0000 Size: 0x4D8000
C:\Windows\SYSTEM32\WINMM.dll Base Address: 0x7FFBD04D0000 Size: 0x27000
C:\Windows\SYSTEM32\WINNSI.DLL Base Address: 0x7FFBD01A0000 Size: 0xB000
C:\Windows\SYSTEM32\WINSPOOL.DRV Base Address: 0x7FFBC2580000 Size: 0x95000
C:\Windows\SYSTEM32\WINSTA.dll Base Address: 0x7FFBD45D0000 Size: 0x5A000
C:\Windows\SYSTEM32\WKSCLI.DLL Base Address: 0x7FFBD4A10000 Size: 0x17000
C:\Windows\SYSTEM32\WTSAPI32.dll Base Address: 0x7FFBD1040000 Size: 0x14000
C:\Windows\SYSTEM32\Wldp.dll Base Address: 0x7FFBD5210000 Size: 0x2E000
C:\Windows\SYSTEM32\XINPUT1_3.dll Base Address: 0x400000 Size: 0x1E000
C:\Windows\SYSTEM32\amsi.dll Base Address: 0x7FFBC9D30000 Size: 0x19000
C:\Windows\SYSTEM32\avrt.dll Base Address: 0x7FFBD0180000 Size: 0xA000
C:\Windows\SYSTEM32\credui.dll Base Address: 0x7FFBD0E10000 Size: 0x11000
C:\Windows\SYSTEM32\d3d10_1core.dll Base Address: 0x7FFBC74E0000 Size: 0xD000
C:\Windows\SYSTEM32\d3d9.dll Base Address: 0x7FFBC5BD0000 Size: 0x1CE000
C:\Windows\SYSTEM32\d3dx9_43.dll Base Address: 0x7FFBB40C0000 Size: 0x263000
C:\Windows\SYSTEM32\dbgcore.DLL Base Address: 0x7FFBD0250000 Size: 0x2C000
C:\Windows\SYSTEM32\dbghelp.dll Base Address: 0x7FFBD02E0000 Size: 0x1E4000
C:\Windows\SYSTEM32\dhcpcsvc.DLL Base Address: 0x7FFBCFEA0000 Size: 0x1D000
C:\Windows\SYSTEM32\dhcpcsvc6.DLL Base Address: 0x7FFBCFF30000 Size: 0x17000
C:\Windows\SYSTEM32\dwmapi.dll Base Address: 0x7FFBD3640000 Size: 0x2F000
C:\Windows\SYSTEM32\dxcore.dll Base Address: 0x7FFBCDB10000 Size: 0x3B000
C:\Windows\SYSTEM32\dxgi.dll Base Address: 0x7FFBD4120000 Size: 0xF4000
C:\Windows\SYSTEM32\dxva2.dll Base Address: 0x7FFBD0860000 Size: 0x24000
C:\Windows\SYSTEM32\gpapi.dll Base Address: 0x7FFBD40F0000 Size: 0x23000
C:\Windows\SYSTEM32\iertutil.dll Base Address: 0x7FFBCB160000 Size: 0x2B0000
C:\Windows\SYSTEM32\inputhost.dll Base Address: 0x7FFBCCF20000 Size: 0x152000
C:\Windows\SYSTEM32\kernel.appcore.dll Base Address: 0x7FFBD3750000 Size: 0x12000
C:\Windows\SYSTEM32\ksuser.dll Base Address: 0x7FFBD0E00000 Size: 0x9000
C:\Windows\SYSTEM32\midimap.dll Base Address: 0x7FFBC6B20000 Size: 0xB000
C:\Windows\SYSTEM32\msacm32.drv Base Address: 0x7FFBC7390000 Size: 0xD000
C:\Windows\SYSTEM32\msdmo.dll Base Address: 0x7FFBC74F0000 Size: 0xB000
C:\Windows\SYSTEM32\ncrypt.dll Base Address: 0x7FFBD5280000 Size: 0x27000
C:\Windows\SYSTEM32\netutils.dll Base Address: 0x7FFBD4D70000 Size: 0xC000
C:\Windows\SYSTEM32\ntdll.dll Base Address: 0x7FFBD8110000 Size: 0x1F5000
C:\Windows\SYSTEM32\ntmarta.dll Base Address: 0x7FFBD4520000 Size: 0x33000
C:\Windows\SYSTEM32\ondemandconnroutehelper.dll Base Address: 0x7FFBBBFB0000 Size: 0x17000
C:\Windows\SYSTEM32\powrprof.dll Base Address: 0x7FFBD56A0000 Size: 0x4B000
C:\Windows\SYSTEM32\profapi.dll Base Address: 0x7FFBD5770000 Size: 0x1F000
C:\Windows\SYSTEM32\resourcepolicyclient.dll Base Address: 0x7FFBD3340000 Size: 0x14000
C:\Windows\SYSTEM32\srvcli.dll Base Address: 0x7FFBC99E0000 Size: 0x28000
C:\Windows\SYSTEM32\textinputframework.dll Base Address: 0x7FFBCD080000 Size: 0xF9000
C:\Windows\SYSTEM32\twinapi.appcore.dll Base Address: 0x7FFBCFBC0000 Size: 0x201000
C:\Windows\SYSTEM32\urlmon.dll Base Address: 0x7FFBC9AF0000 Size: 0x1EB000
C:\Windows\SYSTEM32\vm3dum64_loader.dll Base Address: 0x7FFBCD6D0000 Size: 0x2D000
C:\Windows\SYSTEM32\wdmaud.drv Base Address: 0x7FFBC4FB0000 Size: 0x46000
C:\Windows\SYSTEM32\wevtapi.dll Base Address: 0x7FFBD0960000 Size: 0x65000
C:\Windows\SYSTEM32\windows.storage.dll Base Address: 0x7FFBD3950000 Size: 0x790000
C:\Windows\SYSTEM32\winmmbase.dll Base Address: 0x7FFBBC800000 Size: 0x26000
C:\Windows\SYSTEM32\wintypes.dll Base Address: 0x7FFBD1640000 Size: 0x154000
C:\Windows\SYSTEM32\xinput1_4.dll Base Address: 0x7FFBC7410000 Size: 0x11000
C:\Windows\System32\COMDLG32.dll Base Address: 0x7FFBD75B0000 Size: 0xDA000
C:\Windows\System32\CRYPT32.dll Base Address: 0x7FFBD5D10000 Size: 0x156000
C:\Windows\System32\FWPolicyIOMgr.dll Base Address: 0x7FFBC9CE0000 Size: 0x41000
C:\Windows\System32\FirewallAPI.dll Base Address: 0x7FFBD4450000 Size: 0x9F000
C:\Windows\System32\GDI32.dll Base Address: 0x7FFBD7B80000 Size: 0x2B000
C:\Windows\System32\IMM32.dll Base Address: 0x7FFBD80A0000 Size: 0x30000
C:\Windows\System32\KERNEL32.DLL Base Address: 0x7FFBD7990000 Size: 0xBE000
C:\Windows\System32\KERNELBASE.dll Base Address: 0x7FFBD5A40000 Size: 0x2C8000
C:\Windows\System32\MMDevApi.dll Base Address: 0x7FFBCF770000 Size: 0x85000
C:\Windows\System32\MSCTF.dll Base Address: 0x7FFBD6160000 Size: 0x115000
C:\Windows\System32\NSI.dll Base Address: 0x7FFBD7690000 Size: 0x8000
C:\Windows\System32\OLEAUT32.dll Base Address: 0x7FFBD7FD0000 Size: 0xCD000
C:\Windows\System32\PSAPI.DLL Base Address: 0x7FFBD6300000 Size: 0x8000
C:\Windows\System32\RPCRT4.dll Base Address: 0x7FFBD7A50000 Size: 0x12A000
C:\Windows\System32\SETUPAPI.dll Base Address: 0x7FFBD6470000 Size: 0x472000
C:\Windows\System32\SHELL32.dll Base Address: 0x7FFBD6DA0000 Size: 0x73F000
C:\Windows\System32\SHLWAPI.dll Base Address: 0x7FFBD7550000 Size: 0x55000
C:\Windows\System32\USER32.dll Base Address: 0x7FFBD7C40000 Size: 0x1A1000
C:\Windows\System32\WINTRUST.dll Base Address: 0x7FFBD5F80000 Size: 0x60000
C:\Windows\System32\WS2_32.dll Base Address: 0x7FFBD74E0000 Size: 0x6B000
C:\Windows\System32\WindowManagementAPI.dll Base Address: 0x7FFBCFDD0000 Size: 0xA1000
C:\Windows\System32\Windows.UI.dll Base Address: 0x7FFBCD210000 Size: 0x141000
C:\Windows\System32\advapi32.dll Base Address: 0x7FFBD7EA0000 Size: 0xAC000
C:\Windows\System32\bcrypt.dll Base Address: 0x7FFBD5FE0000 Size: 0x27000
C:\Windows\System32\bcryptPrimitives.dll Base Address: 0x7FFBD5980000 Size: 0x83000
C:\Windows\System32\cfgmgr32.dll Base Address: 0x7FFBD5930000 Size: 0x4E000
C:\Windows\System32\clbcatq.dll Base Address: 0x7FFBD68F0000 Size: 0xA9000
C:\Windows\System32\combase.dll Base Address: 0x7FFBD6A40000 Size: 0x355000
C:\Windows\System32\cryptnet.dll Base Address: 0x7FFBC6490000 Size: 0x31000
C:\Windows\System32\fwbase.dll Base Address: 0x7FFBD43E0000 Size: 0x2F000
C:\Windows\System32\fwpuclnt.dll Base Address: 0x7FFBCE4A0000 Size: 0x7F000
C:\Windows\System32\gdi32full.dll Base Address: 0x7FFBD5E70000 Size: 0x10B000
C:\Windows\System32\imagehlp.dll Base Address: 0x7FFBD62E0000 Size: 0x1D000
C:\Windows\System32\msvcp_win.dll Base Address: 0x7FFBD6010000 Size: 0x9D000
C:\Windows\System32\msvcrt.dll Base Address: 0x7FFBD69A0000 Size: 0x9E000
C:\Windows\System32\ole32.dll Base Address: 0x7FFBD76B0000 Size: 0x12A000
C:\Windows\System32\sechost.dll Base Address: 0x7FFBD6310000 Size: 0x9B000
C:\Windows\System32\shcore.dll Base Address: 0x7FFBD7DF0000 Size: 0xAD000
C:\Windows\System32\ucrtbase.dll Base Address: 0x7FFBD5830000 Size: 0x100000
C:\Windows\System32\win32u.dll Base Address: 0x7FFBD5A10000 Size: 0x22000
C:\Windows\System32\winrnr.dll Base Address: 0x7FFBC7550000 Size: 0x12000
C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.19041.1110_none_792d1c772443f647\COMCTL32.dll Base Address: 0x7FFBB6940000 Size: 0xB0000
C:\Windows\system32\NLAapi.dll Base Address: 0x7FFBD1620000 Size: 0x1D000
C:\Windows\system32\XAudio2_7.dll Base Address: 0x7FFBB5010000 Size: 0x8B000
C:\Windows\system32\napinsp.dll Base Address: 0x7FFBC7570000 Size: 0x17000
C:\Windows\system32\rsaenh.dll Base Address: 0x7FFBD48F0000 Size: 0x34000
C:\Windows\system32\uxtheme.dll Base Address: 0x7FFBD31D0000 Size: 0x9E000
C:\Windows\system32\vm3dum64_10.dll Base Address: 0x7FFBCD1A0000 Size: 0x69000
C:\Windows\system32\wshbth.dll Base Address: 0x7FFBC4240000 Size: 0x15000

Dumping registers:

RAX: 0xB11CBFCAF0
RCX: 0xB11CBFCAF0
RDX: 0x0
RBX: 0xB11CBFD130
RSI: 0x7FFBAE746750
RDI: 0x16
RSP: 0xB11CBFCAC0
RBP: 0xB11CBFD0E0
R8:  0x4D0
R9:  0x0
R10: 0x12
R11: 0xB11CBFCAF0
R12: 0x7FFBD5920920
R13: 0x0
R14: 0x0
R15: 0x910

stack dump [0]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0xCAF80 0x7FFBAE74AF80
stack dump [1]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0xC6793 0x7FFBAE746793
stack dump [2]	 C:\Windows\System32\ucrtbase.dll+0x71881 raise
stack dump [3]	 C:\Windows\System32\ucrtbase.dll+0x72851 abort
stack dump [4]	 C:\Windows\System32\ucrtbase.dll+0x71F9F terminate
stack dump [5]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x9D5B3 0x7FFBAE71D5B3
stack dump [6]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x4F546 0x7FFBAE6CF546
stack dump [7]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x10EA 0x7FFBC7EC10EA
stack dump [8]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x2CA6 _NLG_Return2
stack dump [9]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x21FD _NLG_Return2
stack dump [10]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x40E9 _CxxFrameHandler4
stack dump [11]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x12B1E4 0x7FFBAE7AB1E4
stack dump [12]	 C:\Windows\SYSTEM32\ntdll.dll+0xA214F _chkstk
stack dump [13]	 C:\Windows\SYSTEM32\ntdll.dll+0x30939 RtlUnwindEx
stack dump [14]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x3D76 _NLG_Return2
stack dump [15]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x1635 _NLG_Return2
stack dump [16]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x19E6 _NLG_Return2
stack dump [17]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x232B _NLG_Return2
stack dump [18]	 C:\Windows\SYSTEM32\VCRUNTIME140_1.dll+0x40E9 _CxxFrameHandler4
stack dump [19]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x12B1E4 0x7FFBAE7AB1E4
stack dump [20]	 C:\Windows\SYSTEM32\ntdll.dll+0xA20CF _chkstk
stack dump [21]	 C:\Windows\SYSTEM32\ntdll.dll+0x51454 RtlRaiseException
stack dump [22]	 C:\Windows\SYSTEM32\ntdll.dll+0x511A5 RtlRaiseException
stack dump [23]	 C:\Windows\System32\KERNELBASE.dll+0x34F69 RaiseException
stack dump [24]	 C:\Windows\SYSTEM32\VCRUNTIME140.dll+0x6220 CxxThrowException
stack dump [25]	 C:\Windows\SYSTEM32\MSVCP140.dll+0x39DB2 std::_Xout_of_range
stack dump [26]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x874EF 0x7FFBAE7074EF
stack dump [27]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x85A16 0x7FFBAE705A16
stack dump [28]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x4B2DE 0x7FFBAE6CB2DE
stack dump [29]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x500EA 0x7FFBAE6D00EA
stack dump [30]	 C:\Users\Administrator\Desktop\YimMenu (8).dll+0x463D1 0x7FFBAE6C63D1
stack dump [31]	 C:\Windows\System32\KERNEL32.DLL+0x17034 BaseThreadInitThunk
stack dump [32]	 C:\Windows\SYSTEM32\ntdll.dll+0x52651 RtlUserThreadStart

Exiting after fatal event  (FATAL_SIGNAL). Fatal type:  SIGABRT
Log content flushed successfully to sink

### Expected Behavior

Crash of injection

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I used YimMenu unmodified

### PR DESCRIPTION
**Pull request by maybegreat48**
## Features
* Lessen breakup kicks as host. Disabled by default as someone warned me that it may cause AC detections
* Fast join (like really fast) without forcing script host
* Spoof names of remote players when host
* Advertise YimMenu by renaming everyone's names to differently colored "YimMenu"s when host
* Join players CEO/MC
* Start events on the lobby
* Block malicious launcher events
* Remote control players vehicles (will remove if considered too toxic)
* Block chat advertisements
* Sound crash protection
* Remote off radar/never wanted/semi godmode for individual players or for the entire lobby (it is not recommended to enable all three of them at once for the entire lobby)
* Blame explosion detection
* Copy selected player's RID and IP address

## Fixes
* Fixed player list when multiple players have the same name
* Fixed FPS drops when players with desync protections are present
* Fixed crashes when loading players from database
* Fixed player flickering if invisibility is enabled in online
* Fixed not being able to join another YimMenu user when force session host is enabled for both

Fixes #620 
Closes #234
Closes #178

### CODE REVIEW

**Comment by tupoy-ya** [comment]
> Advertise YimMenu by renaming everyone's names to differently colored "YimMenu"s when host

The next step would be to start spamming Chinese YimMenu advertisements in chat.

File: src/services/players/player_service.hpp [hunk 1]
```diff
@@ -6,8 +6,8 @@ namespace big
 	class player;
 
 	using player_ptr = std::shared_ptr<player>;
-	using player_entry = std::pair<std::string, player_ptr>;
-	using players = std::map<std::string, player_ptr>;
+	using player_entry = std::pair<std::uint8_t, player_ptr>;
+	using players = std::map<std::uint8_t, player_ptr>;
```
Comment (Yimura) [inline src/services/players/player_service.hpp]: This change will remove the alphabetical ordering on the player list, I looked around but did not see you do anything else to order it alphabetically.

File: src/services/players/player_service.cpp [hunk 2]
```diff
@@ -49,15 +40,15 @@ namespace big
 
 	player_ptr player_service::get_by_id(uint32_t id) const
 	{
-		for (const auto& [name, player] : m_players)
-			if (player->id() == id)
+		for (const auto& [pid, player] : m_players)
+			if (pid == id)
```
Comment (Yimura) [inline src/services/players/player_service.cpp]: Can't you just `m_players.find(id)`?

File: src/services/players/player_service.hpp (hunk 1 above)
Comment (maybegreat48) [reply src/services/players/player_service.hpp]: The only easy way to fix this is to revert back to string keys and switching to `std::multimap`. Looking for alternative solutions however

**Comment by Vali0004** [comment]
Block chat advert? :pog: Finally lmao

**Review by Yimura — APPROVED** [review]
No further issues found.

**Pull request merged**